modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionsEvictManager.java

```diff
@@ -657,13 +664,20 @@ private PartitionEvictionTask(
                 return;
             }
 
-            BooleanSupplier stopClo = () -> grpEvictionCtx.shouldStop() || (state.get() == Boolean.FALSE);
+            BooleanSupplier stopClo = () -> {
+                boolean stop = grpEvictionCtx.shouldStop() || (state.get() == Boolean.FALSE);
+
+                if (!stop)
+                    showProgress();
```

Comment:
It seems a bit esoteric to use a 'whether we should stop' closure to print progress. Can an explicit abstraction like `ProgressListener` be used here?

Reply:
It is possible, but I felt like it would add unnecessary noise to the `clearAll` call